### PR TITLE
Rebuild the templates when the instance id change

### LINKED ISSUE #1583: a modif in the .mk doesnt always trigger a template task during the build
when modifing the .mk and executing the make build, the "--engine template" task isnt triggered,
I have to either execute the make twice or touch one of the .in

### PR DESCRIPTION
**Pull request by sbrunner**
Partially fix #1583

### CODE REVIEW

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl [hunk 1]
```diff
@@ -785,6 +782,12 @@ $(APACHE_CONF_DIR)/$(INSTANCE_ID).conf:
 UPGRADE_MAKE_FILE ?= $(INSTANCE_ID).mk
 UPGRADE_ARGS ?=
 
+# instance id marker
+INSTANCE_ID_MARKER ?= .build/instanceid-$(INSTANCE_ID).timestamp
+$(INSTANCE_ID_MARKER):
+	rm -f .build/instanceid-*.timestamp | true
```
Comment (elemoine) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: What does the `| true` do?

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl [hunk 2]
```diff
@@ -423,10 +422,8 @@ endif
 
 $(VENV_BIN)/c2c-template: .build/dev-requirements.timestamp
 
-.build/config-$(INSTANCE_ID).timestamp: $(VENV_BIN)/c2c-template $(VARS_DEPENDS) $(MAKE_FILES)
-	rm -f .build/config-*.timestamp | true
+.build/config.yaml: $(INSTANCE_ID_MARKER) $(VENV_BIN)/c2c-template $(VARS_DEPENDS) $(MAKE_FILES)
 	$(C2C_TEMPLATE_CMD) --get-config .build/config.yaml $(CONFIG_VARS)
```
Comment (elemoine) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: You could replace `.build/config.yaml` with `$@` here.

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: don't failed if there isn't any files to remove :-)

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl (hunk 2 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: +1, thanks :-)

**Pull request merged**